src/functions/definitions/connector_function.ts

```diff
@@ -0,0 +1,46 @@
+import {
+  ManifestFunctionSchema,
+  ManifestFunctionType,
+} from "../../manifest/manifest_schema.ts";
+import {
+  ParameterSetDefinition,
+  PossibleParameterKeys,
+} from "../../parameters/types.ts";
+import { FunctionDefinitionArgs } from "../types.ts";
+import { FunctionDefinition } from "./base_function.ts";
+
+export class ConnectorFunctionDefinition<
+  InputParameters extends ParameterSetDefinition,
+  OutputParameters extends ParameterSetDefinition,
+  RequiredInput extends PossibleParameterKeys<InputParameters>,
+  RequiredOutput extends PossibleParameterKeys<OutputParameters>,
+> extends FunctionDefinition<
+  InputParameters,
+  OutputParameters,
+  RequiredInput,
+  RequiredOutput
+> {
+  type: ManifestFunctionType = "API";
+  constructor(
+    public definition: FunctionDefinitionArgs<
+      InputParameters,
+      OutputParameters,
+      RequiredInput,
+      RequiredOutput
+    >,
+  ) {
+    super(definition);
+  }
+
+  export(): ManifestFunctionSchema {
+    return {
+      title: this.definition.title,
+      description: this.definition.description,
+      type: this.type,
+      input_parameters: this.definition.input_parameters ??
+        { properties: {}, required: [] },
+      output_parameters: this.definition.output_parameters ??
+        { properties: {}, required: [] },
+    };
+  }
```

Comment:
Ah 🤔  the reason why `export()` is in the abstract base class and `ConnectorFunctions` implements it is because `addStep` expects [ISlackFunctionDefinition](https://github.com/slackapi/deno-slack-sdk/blob/156def3d584b6b66a8e3935761b508188d8494fe/src/functions/types.ts#LL302C30-L302C30) as an [argument](https://github.com/slackapi/deno-slack-sdk/blob/156def3d584b6b66a8e3935761b508188d8494fe/src/workflows/mod.ts#L119)

We can potentially change this but it will be a bigger refactor then I expected, LMK what you think

Reply:
Hmm, are there negative repercussions making `export` in `ISlackFunctionDefinition` optional?

Reply:
We would have to add the following where ever `export` or `registerParameterTypes` is used

```ts
if (fn.export) {
}

if (fn.registerParameterTypes) {
}
```

They seem to only be used in 2 places, maybe this is worth implementing? lmk what you think

Reply:
That seems acceptable to me!